### PR TITLE
{lib}[foss/2022b] igraph v0.10.6, leidenalg v0.10.1, libleidenalg v0.11.1, python-igraph v0.10.6

### PR DESCRIPTION
**Pull request by fizwit**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/l/leidenalg/leidenalg-0.10.1-foss-2022b.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+dependencies = [
+    ('Python', '3.10.8'),
+    ('igraph', '0.10.6'),
+    ('python-igraph', '0.10.6'),
+    ('libleidenalg', '0.11.1'),
+]
+
+use_pip = True
+sanity_pip_check = True
+
```
Comment (casparvl) [inline easybuild/easyconfigs/l/leidenalg/leidenalg-0.10.1-foss-2022b.eb:30]: I see there was a 

```
local_preinstallopts = "python setup.py build --use-pkg-config && "
```

In leidenalg-0.9.1-foss-2022a.eb. Can I assume that you removed that intentionally and that it is not needed anymore? (I'm not sure why it was there, there is no comment to indicate why it was needed, so it is hard to say if it still is)

File: easybuild/easyconfigs/l/libleidenalg/libleidenalg-0.11.1-foss-2022b.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    ('CMake', '3.24.3'),
+    ('binutils', '2.39'),
+]
+
+dependencies = [
+    ('igraph', '0.10.6'),
+]
+
+sanity_check_paths = {
+    'files': ['include/%(name)s/GraphHelper.h', 'include/%(name)s/libleidenalg_export.h', 'lib/lib%(name)s.so'],
```
Comment (casparvl) [inline easybuild/easyconfigs/l/libleidenalg/libleidenalg-0.11.1-foss-2022b.eb:25]: Please use SHLIB_EXT (see https://docs.easybuild.io/version-specific/easyconfig-templates/#template-constants-that-can-be-used-in-easyconfigs) instead of hardcodeing the `.so` extension.

File: easybuild/easyconfigs/p/python-igraph/python-igraph-0.10.6-foss-2022b.eb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+exts_list = [
+    ('texttable', '1.6.4', {
+        'checksums': ['42ee7b9e15f7b225747c3fa08f43c5d6c83bc899f80ff9bae9319334824076e9'],
+    }),
+    ('cairocffi', '1.6.1', {
+        'checksums': ['78e6bbe47357640c453d0be929fa49cd05cce2e1286f3d2a1ca9cbda7efdb8b7'],
+    }),
+    # Project name 'python-igraph' is depreicated, use pypi package igraph instead. To avoid
+    # name collision within EB, continue to use the package name 'python-igraph',
+    # to avoid collision with igraph.
```
Comment (casparvl) [inline easybuild/easyconfigs/p/python-igraph/python-igraph-0.10.6-foss-2022b.eb:36]: Maybe rephrase to 'to avoid collision with the C/C++ build of igraph'. I had to look up if the `igraph` this collided with was _actually_ a different package. Rephrasing it that way would indicates that it _is_ the same package, but a different interface.

**Review by casparvl — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/l/leidenalg/leidenalg-0.10.1-foss-2022b.eb (hunk 1 above)
Comment (fizwit) [reply easybuild/easyconfigs/l/leidenalg/leidenalg-0.10.1-foss-2022b.eb:30]: I do not know why "--use-pkg-config" was in the old. Seems to build without it.

File: easybuild/easyconfigs/p/python-igraph/python-igraph-0.10.6-foss-2022b.eb (hunk 3 above)
Comment (fizwit) [reply easybuild/easyconfigs/p/python-igraph/python-igraph-0.10.6-foss-2022b.eb:36]: does this help?
```
# The pypi project 'python-igraph' is depreicated, use pypi package igraph instead. To avoid
# to avoid collision with the C/C++ build of igraph, continue to use the package name 'python-igraph'

File: easybuild/easyconfigs/l/libleidenalg/libleidenalg-0.11.1-foss-2022b.eb (hunk 2 above)
Comment (fizwit) [reply easybuild/easyconfigs/l/libleidenalg/libleidenalg-0.11.1-foss-2022b.eb:25]: ok

File: easybuild/easyconfigs/p/python-igraph/python-igraph-0.10.6-foss-2022b.eb (hunk 3 above)
Comment (casparvl) [reply easybuild/easyconfigs/p/python-igraph/python-igraph-0.10.6-foss-2022b.eb]: Perfect!

File: easybuild/easyconfigs/l/leidenalg/leidenalg-0.10.1-foss-2022b.eb (hunk 1 above)
Comment (casparvl) [reply easybuild/easyconfigs/l/leidenalg/leidenalg-0.10.1-foss-2022b.eb:30]: Ok, let's go without it. If it was there for a reason, I'm sure we'll hit it again - and then hopefully document properly what it fixed :P

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
@boegelbot please test @ jsc-zen2

**Comment by casparvl** [comment]
Test report by @casparvl
**FAILED**
Build succeeded for 3 out of 6 (4 easyconfigs in total)
tcn1.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/5541848347fab625cccdac9cbf1ade29 for a full test report.

**Comment by casparvl** [comment]
Ugh, that failure is not related to this PR at all. See https://github.com/easybuilders/easybuild-easyconfigs/pull/18509#issuecomment-1680579520 

Once I fix it, I can retry a build. But, maybe another maintainer can just upload an extra test report. I think with that, we have sufficient tests here for this to be merged...

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
tcn1.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/c46fcb466ad3c77abce947d61b218b61 for a full test report.

**Review by casparvl — APPROVED** [review]

**Comment by casparvl** [comment]
Going in, thanks @fizwit!
